### ISSUE TITLE
Inverted button_up() and button_down() signal in the _input() function

### ISSUE THREAD

**Issue by kreaninw** [godotengine/godot#59717]
### Godot version

v4.0.alpha5.official [d7d528c15]

### System information

Windows 11, Vulkan Clustered, Intel UHD Graphics 620 v30.0.101.1631

### Issue description

I connect 2 signals, `button_up()` and `button_down()`, from the `TextureButton` node, as shown in the screenshot below:

<img width="389" alt="Screenshot 2022-03-31 001723" src="https://user-images.githubusercontent.com/1403194/160971801-e0d28e37-ccfb-4cac-90d8-1586c95b7377.png">

Then, I print the variable above in the `_input()` function when the mouse button is downed, as shown in the screenshot below:

<img width="613" alt="Screenshot 2022-03-31 001922" src="https://user-images.githubusercontent.com/1403194/160972028-f5b8b0e0-f94d-43f5-aaa1-5f4b25450c09.png">

However, it seems the result is inverted as shown in the screen recording below:

https://user-images.githubusercontent.com/1403194/160972225-a4318224-f576-42a4-81a3-1f906d944fab.mp4

But I check the variable value in the `_process()` function, the signal is working fine as shown in the screen recording below:

https://user-images.githubusercontent.com/1403194/160972507-948d0801-517d-4076-bfd4-0b3710fb71ed.mp4

The event as reported from `InputEventMouseButton` is also contrary to the result from the signal, as shown in the screen recording below:

https://user-images.githubusercontent.com/1403194/160974931-e26d9b13-36ad-40f3-9cba-cacb0a4a8338.mp4

I expect the signal to work correctly in the `_input()` function.

### Steps to reproduce

1. Open the minimal reproduction project.
2. Run the project.
3. Click anywhere on the screen (hold down and release LMB) and see the result in print. The result will be inverted.

### Minimal reproduction project

[Test.zip](https://github.com/godotengine/godot/files/8385889/Test.zip)

**Comment by timothyqiu** [comment]
These signals are emitted after `_input()`. So what happens when you press down the mouse button is:

1. `_input()` is called, the last value of the `button_down` variable is printed, i.e. `false`.
2. `button_down()` signal is emitted, `button_down` is set to `true`.

**Comment by madmiraal** [comment]
> But I don't know what's the point of these input signals if they don't fire at `_input()` time. For instance, when the user holds down LMB, the `button_down()` signal would do nothing at all, and the actions/functions set in the `button_down()` signal will be fired when the user actually released LMB, which is very counterintuitive.

The `button_down()` signal does fire when the user presses the left mouse button. Add a print statement to your `_on_texture_button_down()` function to confirm this. As @timothyqiu points out, all you are seeing is that the `_input()` function is called before the signal is generated and your `_on_texture_button_down()` function is called. This is by design. See the [documentation](https://docs.godotengine.org/en/stable/tutorials/inputs/inputevent.html) to understand which order the functions are called.

> However, this doesn't happen if you change the TypeCast to InputEventScreenTouch (to test with touch screen) instead of InputEventMouseButton.

The project setting: `emulate_mouse_from_touch` is enabled by default, so a screen touch will generate an `InputEventMouseButton` event, call `_input(InputEventMouseButton)`, trigger the signal and call `_on_texture_button_down()`, and only then call `_input(InputEventScreenTouch)`. Again the only question is the order the functions are called, not whether or not the functions are called.

In summary, this is not a bug. You should either use the `button_down` signal or the `_input()`, `_gui_input()` or `_unhandled_input()` functions, not both. If you have any further questions, please one of the community forums: https://godotengine.org/community

**Comment by kreaninw** [comment]
Close as it's explained in the comment above. 

Thanks @madmiraal
